code/modules/awaymissions/mission_code/ruins/watcher_grave.dm

```diff
@@ -0,0 +1,256 @@
+	RegisterSignal(orbiter, COMSIG_PARENT_QDELETING, PROC_REF(our_remove_orbiter))
+
+/obj/item/watcher_hatchling/Moved(atom/old_loc, movement_dir, forced)
+	. = ..()
+	if(isnull(orbiter))
+		return
+	var/mob/holder = get(loc, /mob)
+	if(holder != owner)
+		watcher_return()
+
+/// If the guy we are orbiting is deleted but somehow we aren't
+/obj/item/watcher_hatchling/proc/remove_owner()
+	SIGNAL_HANDLER
+	UnregisterSignal(owner, COMSIG_PARENT_QDELETING)
+	owner = null
+
+/// In the more likely event that our orbiter is deleted, stop holding a reference to it
+/obj/item/watcher_hatchling/proc/our_remove_orbiter()
+	SIGNAL_HANDLER
+	orbiter = null // No need to unregister signal because we only call this when it deletes
+
+/// Get back in your ball pikachu
+/obj/item/watcher_hatchling/proc/watcher_return()
+	qdel(orbiter)
+	remove_owner()
+
+
+/// Orbiting visual which shoots at mining mobs
+/obj/effect/watcher_orbiter
+	name = "watcher hatchling"
+	icon = 'icons/mob/lavaland/lavaland_monsters.dmi'
+	icon_state = "watcher_baby"
+	layer = EDGED_TURF_LAYER // Don't render under lightbulbs
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	pixel_y = 22
+	alpha = 0
+	/// Who are we following?
+	var/atom/parent
+	/// Datum which keeps us hanging out with our parent
+	var/datum/movement_detector/tracker
+	/// Type of projectile we fire
+	var/projectile_type = /obj/item/projectile/baby_watcher_blast
+	/// Sound to make when we shoot
+	var/projectile_sound = 'sound/weapons/pierce.ogg'
+	/// Time between taking potshots at goliaths
+	var/fire_delay = 5 SECONDS
+	/// How much faster do we shoot when avenging our parent?
+	var/on_death_multiplier = 5
+	/// Time taken between shots
+	COOLDOWN_DECLARE(shot_cooldown)
+	/// Types of mobs to attack
+	var/list/target_faction = list("hostile", "terrorspiders", "mining", "boss") //Yes terror spiders are their own faction. Also fuck them.
+
+/obj/effect/watcher_orbiter/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+// Shuttle rotation fucks with our position, we just want to stick with our guy
+/obj/effect/watcher_orbiter/shuttleRotate(rotation, params)
+	return
+
+/obj/effect/watcher_orbiter/Destroy(force)
+	STOP_PROCESSING(SSobj, src)
+	QDEL_NULL(tracker)
+	return ..()
+
+/obj/effect/watcher_orbiter/process(seconds_per_tick)
+	if(!COOLDOWN_FINISHED(src, shot_cooldown))
+		return
+	for(var/mob/living/potential_target in oview(5, src))
+		if(!isanimal(potential_target) || potential_target.stat == DEAD)
+			continue
+		if(!faction_check(target_faction, potential_target.faction))
+			continue
+		shoot_at(potential_target)
+		return
+
+/// Take a shot
+/obj/effect/watcher_orbiter/proc/shoot_at(atom/target)
+	COOLDOWN_START(src, shot_cooldown, fire_delay)
+	var/turf/T = get_turf(src)
+	var/turf/U = get_turf(target)
+	if(!T || !U)
+		return
+	var/obj/item/projectile/O = new projectile_type(T)
+	playsound(get_turf(src), projectile_sound, 75, 1)
+	O.firer = parent // no hitting owner.
```

Comment:
Will this show an attacklog that the parent shot someone? Probably fine, just trying to think of possible admin situations in this case.

Reply:
Yeah, should be fine as it only attacks animals